### PR TITLE
Feature/bug 1138550 unified clones

### PR DESCRIPTION
**Pull request by Pike**

### CODE REVIEW

File: apps/pushes/management/commands/fork.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    def add_arguments(self, parser):
+        parser.add_argument('--dry-run', '-n', action='store_true',
+                            help='Dry run, do not change db or repos')
+        parser.add_argument('orig',
+                            help='Repository or Forest to fork')
+        parser.add_argument('fork',
+                            help='Forked Repository or Forest')
+
+    def handle(self, orig, fork, dry_run=False, **options):
+        import hglib
```
Comment (mathjazz) [inline apps/pushes/management/commands/fork.py]: Just curious, do you need to `import hglib` here?

File: apps/pushes/management/commands/fork.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                raise CommandError(
+                    '%s is neither Repository nor Forest' % orig)
+        try:
+            fork = Repository.objects.get(name=fork)
+        except Repository.DoesNotExist:
+            try:
+                fork = Forest.objects.get(name=fork)
+            except Forest.DoesNotExist:
+                raise CommandError(
+                    '%s is neither Repository nor Forest' % fork)
```
Comment (mathjazz) [inline apps/pushes/management/commands/fork.py]: These two `try - except` blocks call for code reusability.

File: apps/life/models.py [hunk 3]
```diff
@@ -183,6 +209,28 @@ def last_known_push(self):
...
+    def local_path(self):
+        '''If this is a unified local clone, find out which repo
+        we're pulling for that.
+        Used to compute where to search for the local clone, thus
+        the name.
+        '''
+        return os.path.join(
+            settings.REPOSITORY_BASE,
+            *self.relative_path().split('/'))
+
```
Comment (mathjazz) [inline apps/life/models.py]: The `Repository` and `Forest` class could probably inherit from the same parent class.

File: apps/pushes/management/commands/fork.py (hunk 1 above)
Comment (Pike) [inline apps/pushes/management/commands/fork.py]: Legacy coding style. Back in the days, django was a just part of the repo, so you could run at least ./manage.py help without having python dependencies installed.

As django itself is now in the virtualenv, this is obsolete, I moved the import to the top-level.

Might make sense to do that globally, but that's a follow-up.

File: apps/pushes/management/commands/fork.py (hunk 2 above)
Comment (Pike) [inline apps/pushes/management/commands/fork.py]: Done in a follow-up commit, good catch.

File: apps/life/models.py (hunk 3 above)
Comment (Pike) [inline apps/life/models.py]: I entertained the idea for a bit, but then found https://docs.djangoproject.com/en/1.8/topics/db/models/#be-careful-with-related-name, and didn't want to spend actual time on figuring out if this bites me or not.

**Pull request merged**